### ISSUE TITLE
Add option to skip serialization of slendr configuration files

### ISSUE THREAD

**Issue by bodkan** [bodkan/slendr#97]
With the awesome work that @juledrewa has been doing on automated model generation that will be merged into the main branch soon, I think it makes sense to add the possibility to skip the serialization of compiled slendr files to disk (which can be costly in terms of time and space, especially for many many models compiled in parallel), and simply create an R slendr model object that's not backed by config files.

One idea we had was to try to implement a method for quick model fitting via msprime simulation. Unlike the SLiM back end, with Python we have the advantage to call the msprime slendr backend directly from R, without having to go through the command-line and config files. For these use cases, compiling files to disk is not necessary. Doing simple `model <- compile_model(..., serialize = FALSE)` or such would be nice because we could go directly to `msprime(model, ...)` without having to compile a bunch of files to disk first (which could be annoying for thousands/millions of models that might have to be compiled).

Just making a note of that so that I don't forget, and also so that @juledrewa and @FerRacimo know about this.

**Comment by bodkan** [comment]
Actually, in this situation `msprime(model, ...)` could directly return an R-based tskit tree sequence object, without having to save it to disk first in order to be loaded by `ts_load()` after the simulation...

**Comment by bodkan** [comment]
Closed [here](https://github.com/bodkan/slendr/pull/112).
